OSDOCS-19410 back porting most recent zero-trust docs to 4.18 PR#1 (#110828)
openshift/openshift-docs: modules/zero-trust-manager-spire-server-config.adoc

apiVersion: operator.openshift.io/v1alpha1
kind: SpireServer
metadata:

>  name: cluster

Reviewer: ```suggestion
  name: cluster
```

Author: Done

  persistence:
    size: "5Gi"
    accessMode: "ReadWriteOnce"
    storageClass: "gp3-csi"
  datastore:
    databaseType: "sqlite3"
    connectionString: "/run/spire/data/datastore.sqlite3"
    tlsSecretName: ""
    maxOpenConns: 100
    maxIdleConns: 10
    connMaxLifetime: 0
    disableMigration: "false"
----
where:

`metadata.name`:: Specifies that the value must be `cluster`.

`spec.logLevel`:: Specifies the logging level for the SPIRE Server. The valid options are `debug`, `info`, `warn`, and `error`.

`spec.logFormat`:: Specifies the logging format for the SPIRE Server. The valid options are `text` and `json`.

`spec.jwtIssuer`:: Specifies the JWT issuer URL. Must be a valid HTTPS or HTTP URL with a maximum length of 512 characters.

`spec.caValidity`:: Specifies the validity period (Time to Live (TTL)) for the SPIRE Server's CA certificate. This determines how long the server's root or intermediate certificate is valid. The format is a duration string (for example, `24h`, `168h`).

`spec.defaultX509Validity`:: Specifies the default validity period (TTL) for X.509 SVIDs issued to workloads. This value is used if a specific TTL is not configured for a registration entry.

> `spec.defaultJWTValidity`:: Specifies thedefault validity period (TTL) for JWT SVIDs issued to workloads. This value is used if a specific TTL is not configured for a registration entry.

Reviewer: ```suggestion
`spec.defaultJWTValidity`:: Specifies the default validity period (TTL) for JWT SVIDs issued to workloads. This value is used if a specific TTL is not configured for a registration entry.
```

Author: Done

> `spec.wtKeyType`:: Specifies the key type used for JWT signing. The valid options are `rsa-2048`, `rsa-4096`, `ec-p256`, and `ec-p384`. This field is optional.

Reviewer: ```suggestion
`spec.wtKeyType`:: Optional: Specifies the key type used for JWT signing. The valid options are `rsa-2048`, `rsa-4096`, `ec-p256`, and `ec-p384`.
```

Author: Done

`spec.caSubject.organization`:: Specifies the organization for the SPIRE Server CA. Maximum length is 64 characters.

`spec.caSubject.commonName`:: Specifies the common name for the SPIRE Server CA. Maximum length is 255 characters.

`spec.persistence.size`:: Specifies the size of the persistent volume (for example, `1Gi`, `5Gi`). Once set, this field is immutable.

`spec.persistence.accessMode`:: Specifies the access mode for the persistent volume. The valid options are `ReadWriteOnce`, `ReadWriteOncePod`, and `ReadWriteMany`. Once set, this field is immutable.

`spec.persistence.storageClass`::  Specifies the storage class to be used for the PVC. Once set, this field is immutable.

`spec.datastore.databaseType`:: Specifies the type of database to use for the datastore. The valid options are `sql`, `sqlite3`, `postgres`, `mysql`, `aws_postgresql`, and `aws_mysql`.

`spec.datastore.connectionString`:: Specifies the connection string for the database. For PostgreSQL with SSL, include `sslmode` and certificate paths (for example, `dbname=spire user=spire host=postgres.example.com sslmode=verify-full`).

> `spec.datastore.tlsSecretName`::  Specifies the name of a Kubernetes Secret containing TLS certificates for database connections. The Secret will be mounted at `/run/spire/db/certs`. This field is optional.

Reviewer: ```suggestion
`spec.datastore.tlsSecretName`::  Optional: Specifies the name of a Kubernetes Secret containing TLS certificates for database connections. The Secret will be mounted at `/run/spire/db/certs`. 
```

Author: Done

    tlsSecretName: ""
    maxOpenConns: 100
    maxIdleConns: 10
    connMaxLifetime: 0
    disableMigration: "false"
----
where:

`metadata.name`:: Specifies that the value must be `cluster`.

`spec.logLevel`:: Specifies the logging level for the SPIRE Server. The valid options are `debug`, `info`, `warn`, and `error`.

`spec.logFormat`:: Specifies the logging format for the SPIRE Server. The valid options are `text` and `json`.

`spec.jwtIssuer`:: Specifies the JWT issuer URL. Must be a valid HTTPS or HTTP URL with a maximum length of 512 characters.

`spec.caValidity`:: Specifies the validity period (Time to Live (TTL)) for the SPIRE Server's CA certificate. This determines how long the server's root or intermediate certificate is valid. The format is a duration string (for example, `24h`, `168h`).

`spec.defaultX509Validity`:: Specifies the default validity period (TTL) for X.509 SVIDs issued to workloads. This value is used if a specific TTL is not configured for a registration entry.

`spec.defaultJWTValidity`:: Specifies the default validity period (TTL) for JWT SVIDs issued to workloads. This value is used if a specific TTL is not configured for a registration entry.

> `spec.wtKeyType`:: IOptional: Specifies the key type used for JWT signing. The valid options are `rsa-2048`, `rsa-4096`, `ec-p256`, and `ec-p384`.

Reviewer 2: ```suggestion
`spec.wtKeyType`:: Optional parameter. Specifies the key type used for JWT signing. The valid options are `rsa-2048`, `rsa-4096`, `ec-p256`, and `ec-p384`.
```

Author: Done

> `spec.wtKeyType`:: IOptional: Specifies the key type used for JWT signing. The valid options are `rsa-2048`, `rsa-4096`, `ec-p256`, and `ec-p384`.

Reviewer 2: ```suggestion
`spec.jwtKeyType`:: IOptional: Specifies the key type used for JWT signing. The valid options are `rsa-2048`, `rsa-4096`, `ec-p256`, and `ec-p384`.
```

Author: Done

`spec.caSubject.organization`:: Specifies the organization for the SPIRE Server CA. Maximum length is 64 characters.

`spec.caSubject.commonName`:: Specifies the common name for the SPIRE Server CA. Maximum length is 255 characters.

`spec.persistence.size`:: Specifies the size of the persistent volume (for example, `1Gi`, `5Gi`). Once set, this field is immutable.

`spec.persistence.accessMode`:: Specifies the access mode for the persistent volume. The valid options are `ReadWriteOnce`, `ReadWriteOncePod`, and `ReadWriteMany`. Once set, this field is immutable.

`spec.persistence.storageClass`::  Specifies the storage class to be used for the PVC. Once set, this field is immutable.

`spec.datastore.databaseType`:: Specifies the type of database to use for the datastore. The valid options are `sql`, `sqlite3`, `postgres`, `mysql`, `aws_postgresql`, and `aws_mysql`.

`spec.datastore.connectionString`:: Specifies the connection string for the database. For PostgreSQL with SSL, include `sslmode` and certificate paths (for example, `dbname=spire user=spire host=postgres.example.com sslmode=verify-full`).

> `spec.datastore.tlsSecretName`::  Optional: Specifies the name of a Kubernetes Secret containing TLS certificates for database connections. The Secret will be mounted at `/run/spire/db/certs`.

Reviewer 2: ```suggestion
`spec.datastore.tlsSecretName`:: Optional parameter. Specifies the name of a Kubernetes Secret containing TLS certificates for database connections. The Secret will be mounted at `/run/spire/db/certs`.
```

Author: Done

> `spec.datastore.tlsSecretName`::  Optional parameter. Specifies the name of a Kubernetes Secret containing TLS certificates for database connections. The Secret will be mounted at `/run/spire/db/certs`.

Reviewer 2: ```suggestion
`spec.datastore.tlsSecretName`:: Optional parameter. Specifies the name of a Kubernetes Secret containing TLS certificates for database connections. The Secret will be mounted at `/run/spire/db/certs`.
```

Author: Done

> `spec.datastore.tlsSecretName`::  Optional parameter. Specifies the name of a Kubernetes Secret containing TLS certificates for database connections. The Secret will be mounted at `/run/spire/db/certs`.

Reviewer 2: I still see two empty spaces between `spec.datastore.tlsSecretName`:: and Optional parameter.

Author: Done
